zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -250,13 +247,12 @@ public boolean isEmpty() {
     return isEmpty.get();
   }
 
-  private void assertForeignKeyExists(final ZeebeTransaction transaction, final Object key)
+  private void assertForeignKeysExist(final ZeebeTransaction transaction, final Object... keys)
       throws Exception {
-    if (!consistencyChecksSettings.enableForeignKeys()) {
-      return;
-    }
-    if (key instanceof ContainsForeignKeys containsForeignKey) {
-      foreignKeyChecker.assertExists(transaction, containsForeignKey);
+    for (final var key : keys) {
```

Comment:
:wrench: Could add the feature flag again. Then this would be double checked here and it would avoid the loop here and the decomposition in the called method